### PR TITLE
Select marker & Display empty board

### PR DESCRIPTION
**Pull request by purlak**
## Description
Add Content module to hold all string data and update Cli to display data from Content
Update Cli to take in user input and start a new game
Game initializes a new game with an empty array and displays board

### CODE REVIEW

File: lib/cli.rb [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+    display_text(Content.select_marker)
+    user_input 
   end
 
-  def select_marker
-    "Choose 'X' or 'O' to play game: \n"
-  end
+  def display_text(text)
+    puts text + "\n\n"
+  end
```
Comment (ssunday) [inline lib/cli.rb]: extra white space 👀

File: lib/cli.rb [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+    display_text(Content.welcome)
+    display_text(Content.select_marker)
+    user_input 
   end
 
-  def select_marker
-    "Choose 'X' or 'O' to play game: \n"
-  end
+  def display_text(text)
+    puts text + "\n\n"
```
Comment (ssunday) [inline lib/cli.rb]: you can interpolate with "#{text}\n\n"

File: lib/cli.rb [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+  end 
 
   def clear_screen
     system("clear")
   end
+
+  def user_input 
+    gets.chomp
+    Game.new.empty_board
```
Comment (ssunday) [inline lib/cli.rb]: so you get the input....then make an empty board? 🤔 Seems a bit odd

File: lib/content.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module Content
+	def self.welcome
```
Comment (ssunday) [inline lib/content.rb]: Alignment

File: lib/game.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class Game
+	attr_accessor :board 
+
+	def initialize
+		@board = Array.new(9)
+	end
+
+	def empty_board
```
Comment (ssunday) [inline lib/game.rb]: so empty board is really display board?

File: lib/tictactoe.rb [hunk 6]
```diff
@@ -1,4 +1,6 @@
 require_relative './cli.rb'
+require_relative './content.rb'
+require_relative './game.rb'
 require 'pry'
```
Comment (ssunday) [inline lib/tictactoe.rb]: Pry shouldn't be required

File: lib/game.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+class Game
+	attr_accessor :board 
+
+	def initialize
+		@board = Array.new(9)
```
Comment (naush) [inline lib/game.rb]: Indentation.

File: spec/cli_spec.rb [hunk 8]
```diff
@@ -1,25 +1,64 @@
...
 
   it 'clears screen before printing menu' do
     expect(@cli).to receive(:clear_screen)
+    allow(@cli).to receive(:gets).and_return('nil')
+    @cli.call
+  end
 
+  it 'displays welcome message and marker options' do
+    allow(@cli).to receive(:clear_screen)
+    allow(@cli).to receive(:gets).and_return('nil')
```
Comment (naush) [inline spec/cli_spec.rb]: What does return nil mean here? And why nil?

File: spec/cli_spec.rb [hunk 9]
```diff
@@ -1,25 +1,64 @@
 require_relative '../lib/cli.rb'
+require_relative '../lib/content.rb'
+require 'pry'
```
Comment (naush) [inline spec/cli_spec.rb]: This is probably not used.

**Comment by naush** [comment]
@purnima-nyc You can set up a linter with your editor that runs and formats your code automatically. For Ruby, try install rubocop!

File: spec/cli_spec.rb [hunk 10]
```diff
@@ -1,25 +1,64 @@
...
   before do
+    allow($stdin).to receive(:gets)
     allow($stdout).to receive(:puts)
-
     @cli = Cli.new
   end
 
   it 'clears screen before printing menu' do
     expect(@cli).to receive(:clear_screen)
+    allow(@cli).to receive(:gets).and_return('nil')
```
Comment (ssunday) [inline spec/cli_spec.rb]: If this was really nil, you wouldn't need it to be returned and it wouldn't be in quotes

File: lib/cli.rb (hunk 2 above)
Comment (purlak) [reply lib/cli.rb]: commit 240bf9f926484a874279542f5ac97c443441f1f6

File: lib/cli.rb (hunk 1 above)
Comment (purlak) [reply lib/cli.rb]: commit fb9b15eabc95f9a438326dc0aad4c91dbb6faa8b

File: lib/content.rb (hunk 4 above)
Comment (purlak) [reply lib/content.rb]: commit fb9b15eabc95f9a438326dc0aad4c91dbb6faa8b

File: lib/tictactoe.rb (hunk 6 above)
Comment (purlak) [reply lib/tictactoe.rb]: commit 1876366afbfb624b2a53a59ec620ecbd0dacb427

File: lib/cli.rb (hunk 3 above)
Comment (purlak) [reply lib/cli.rb]: The idea there was to get the user to choose their marker ‘X’ or ‘O’, and then the empty board is displayed, next step would be to play a turn. any thoughts what I could do different there? An idea would be to first display the empty board and then ask the user whether they would like to choose X or O - choosing X can allow the user to have the first turn and start the game.

File: lib/game.rb (hunk 5 above)
Comment (purlak) [reply lib/game.rb]: Yes, the idea here was to allow the user to first view the empty board and then play a turn. But I think it makes sense to use the same method to display the board after every turn/move and rename it to display_board. 
commit 64be04ddd8a7786bd5eda8abfd8fa4256933c192

File: lib/tictactoe.rb (hunk 6 above)
Comment (purlak) [reply lib/tictactoe.rb]: commit 61ba9858a6b50abc5ed9c5e92ac21905d8aa9657

File: spec/cli_spec.rb (hunk 10 above)
Comment (purlak) [reply spec/cli_spec.rb]: @ssunday is it really returning an empty space and `allow(@cli).to receive(:gets).and_return(' ')` should fix this?

Comment (ssunday) [reply spec/cli_spec.rb]: If that's what it's supposed to be doing!

Comment (purlak) [reply spec/cli_spec.rb]: commit 8e131fd9e6d8f8cc6525b407331850489f6d58bb

File: lib/cli.rb (hunk 3 above)
Comment (purlak) [reply lib/cli.rb]: @ssunday I moved user_input from Cli to Game, Game now has a call method that will call user_input and then display_board (empty board to start with).  commit 419cb4933d9da60a8fcd82b535700caf5fd638f5

File: spec/cli_spec.rb [hunk 11]
```diff
@@ -1,25 +1,36 @@
...
+describe 'display_text' do
+  before do
+    @cli = Cli.new
+  end
 
-    expect { @cli.call }.to output(expected).to_stdout
+  it 'takes in a text argument displays text' do
+    text = 'test'
+    expect(@cli).to receive(:display_text).with(text).and_return(text)
+    @cli.display_text(text)
```
Comment (naush) [inline spec/cli_spec.rb:34]: Can you walk me through what this test is intended to test?

File: spec/game_spec.rb [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    it 'calls user input' do
+      expect(@game).to receive(:user_input)
+      @game.call
+    end
+
+    it 'shows an empty board' do
+      allow($stdout).to receive(:puts)
+      expected = "  |  |  \n--------\n  |  |  \n--------\n  |  |  \n"
+      expect { @game.display_board }.to output(expected).to_stdout
+      @game.display_board
```
Comment (naush) [inline spec/game_spec.rb:24]: Why is @game.display_board called twice in this test?

File: spec/spec_helper.rb [hunk 13]
```diff
@@ -44,57 +44,55 @@
...
+  #   # order dependency and want to debug it, you can fix the order by providing
+  #   # the seed, which is printed after each run.
+  #   #     --seed 1234
+  #   config.order = :random
+  #
+  #   # Seed global randomization in this process using the `--seed` CLI option.
+  #   # Setting this allows you to use `--seed` to deterministically reproduce
+  #   # test failures related to randomization by passing the same `--seed` value
+  #   # as the one that triggered the failure.
+  #   Kernel.srand config.seed
```
Comment (naush) [inline spec/spec_helper.rb:97]: What is the purpose of this file? Why is it all commented out?

**Review by naush — APPROVED** [review]
Go ahead and merge this PR, @purnima-nyc. We will revisit some of these feedback when I'm back next week. :)

File: spec/cli_spec.rb (hunk 9 above)
Comment (purlak) [reply spec/cli_spec.rb]: commit 1876366

File: lib/game.rb (hunk 7 above)
Comment (purlak) [reply lib/game.rb]: commit fb9b15e

File: spec/cli_spec.rb (hunk 8 above)
Comment (purlak) [reply spec/cli_spec.rb]: This should be ' ' and not 'nil'. commit 8e131fd
https://github.com/purnima-nyc/ttt-ruby/pull/5#discussion_r274043923

**Pull request merged**